std/traits.d

```diff
@@ -3340,6 +3341,171 @@ enum hasMember(T, string name) = __traits(hasMember, T, name);
     static assert(hasMember!(S, "foo"));
 }
 
+/++
+    Whether the symbol represented by the string, member, is a static member of
+    T.
+  +/
+template isStaticMember(T, string member)
+    if (__traits(hasMember, T, member))
+{
+    import std.meta : AliasSeq;
+    import std.traits : FunctionTypeOf;
+    alias sym = AliasSeq!(__traits(getMember, T, member))[0];
+
+    static if (is(typeof(sym) == function) || is(FunctionTypeOf!(typeof(&sym)) == function))
+        enum bool isStaticMember = __traits(isStaticFunction, sym);
+    else
+        enum bool isStaticMember = __traits(compiles, &sym);
```

Comment:
I'm not sure that it matters in this case, but there are subtle differences between the two, and I've been advised in the past to favor `__traits(compiles, ...)`. If I understand correctly, the main difference is that  `__traits(compiles, ...)` actually checks for compilation, whereas `is(typeof(...))` just checks that the type exists (e.g. there was a discussion on it here with Don: https://issues.dlang.org/show_bug.cgi?id=8339 ). `__traits(compiles, ...)` is also more explicit about what you're doing. So, I generally try to use `__traits(compiles, ...)` instead of `is(typeof(...))`. I confess though that I don't understand the subtleties well enough to know when it really matters.